### PR TITLE
added video note feature, add, delete and update note

### PR DESCRIPTION
**Pull request by JantuDeb**
# Video Note
- add, remove and delete a note from the video.
- refactored some codes.
image

### CODE REVIEW

File: src/pages/video-details/VideoDetails.jsx [hunk 1]
```diff
@@ -1,57 +1,68 @@
...
+import { UPDATE_LIKE_COUNT } from "../../context/videos/video-reducer";
 const VideoDetails = () => {
-  const [video, setVideo] = useState({});
   const [searchparams] = useSearchParams();
   const { addToHistory } = useHistoryVideos();
   const { addToLikes, likedVideos, removeFromLikes } = useLikedVideos();
-
+  const { video, videoDispatch, getVideo, updateViewCount, getNote } =
+    useVideos();
   const videoId = searchparams.get("videoId");
```
Comment (sushil-mod) [inline src/pages/video-details/VideoDetails.jsx:22]: i liked you using usesearchparams

File: src/pages/video-details/VideoDetails.jsx [hunk 2]
```diff
@@ -1,57 +1,68 @@
...
+    title,
+    channelTitle,
+    statistics,
+    description,
+    createdAt,
+    videoURL,
+    note,
+  } = video;
+  const isLiked = likedVideos.some((video) => video._id === videoId);
+  console.log(note);
```
Comment (sushil-mod) [inline src/pages/video-details/VideoDetails.jsx:34]: remove log if not in use

**Review by sushil-mod — APPROVED** [review]
just a comment . For me  no mistake was found . Good code  and also its working nice in UI

File: src/components/VideoNote.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  const [text, setText] = useState(note);
+  const [editing, setEditing] = useState(false);
+  const { addNote, deleteNote } = useVideos();
+  console.log(note);
+  const editClickHandler = () => {
+    if (editing) addNote({ note: text, videoId });
+    else if (note && !editing) {
+      setText(note);
+    }
+    setEditing((v) => !v);
```
Comment (dikshadhote) [inline src/components/VideoNote.jsx:16]: Can use meaningful name instead of v

**Review by dikshadhote — APPROVED** [review]
The code is very well structured.

**Pull request merged**